app/src/main/java/org/keynote/godtools/android/fragment/LanguagesFragment.java

```diff
@@ -63,12 +78,18 @@ public static Fragment newInstance(final boolean primary) {
     @Override
     public void onCreate(@Nullable final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
+        // Allow Fragment to have its own menu
+        setHasOptionsMenu(true);
         final Bundle args = getArguments();
         if (args != null) {
             mPrimary = args.getBoolean(EXTRA_PRIMARY, mPrimary);
         }
-
+        if (savedInstanceState != null) {
+            if (savedInstanceState.containsKey(KEY_SEARCH))
+                mQuery = savedInstanceState.getString(KEY_SEARCH);
```

Comment:
2 comments:
- always use braces for if blocks, it helps guard against accidentally executing a statement outside of an if block. I believe there is a checkstyle rule about this.
- in this case if statements shouldn't be required, you can use the `getString` and `getBoolean` methods with default values to simplify this logic.
```
mQuery = savedInstanceState.getString(KEY_SEARCH, mQuery);
mIsSearchViewOpen = savedInstanceState.getBoolean(KEY_IS_SEARCH_OPEN, mIsSearchViewOpen);
```